rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -83,4 +75,65 @@ Recorder::create_subscription(
   return subscription;
 }
 
+std::future<void> Recorder::launch_topics_discovery(
```

Comment:
why do we need this function? Isn't this function just a wrapper around the lambda and its call?

Reply:
Yes, but it very much improves the readability of `record()`.